docs/content/guides/6.multistore/3.patterns/2.design/4.different-code-per-store.md

```diff
@@ -5,4 +5,677 @@ navigation:
+- To reduce complexity in the parent component
+- To make inheritance more explicit and maintainable
+- When you want to avoid unnecessary props in stores that don't need them
+
+Instead of adding specialized props to the root component that all stores inherit, you can remove them from the root and add them only where needed.
+
+#### Example Workflow
+
+1. **Remove** the specialized props from the parent store component
+2. **Add** the props only to the store specific component
+3. **Update** implementations of the changed component in specific store
+
+#### In Next.js
+
+First, remove the `classNameVariants` prop from the root component:
+
+```tsx
+// /apps/storefront-unified-nextjs/components/decorated-price.tsx
+
+import classNames from 'classnames';
+
+import { useFormatter } from '@/hooks/use-formatter';
+import type { SfProduct } from '@/types';
+
+export interface ClassNameVariants {
+  regular?: string;
+  special?: string;
+}
+
+// Remove prop from parent
+export interface DecoratedPriceProps {
+  className?: string;
+  classNameVariants?: ClassNameVariants; // [!code --]
+  price: SfProduct['price'];
+}
+
+export default function DecoratedPrice({ className, classNameVariants, price, ...rest }: DecoratedPriceProps) { // [!code --]
+export default function DecoratedPrice({ className, price, ...rest }: DecoratedPriceProps) { // [!code ++]
+  const { formatPrice } = useFormatter();
+
+  const regular = price?.value ? formatPrice(price.value) : '';
+  const special = price?.isDiscounted ? formatPrice(price!.regularPrice) : undefined;
+
+  return (
+    price && (
+      <div className={classNames('flex items-baseline gap-x-2', className)} {...rest}>
+        <span
+          className={classNames('font-semibold', classNameVariants.regular, { // [!code --]
+          className={classNames('font-semibold', { // [!code ++]
+            'text-neutral-900': !special,
+            'text-secondary-700': special,
+          })}
+          data-testid="special-price"
+        >
+          {regular}
+        </span>
+        {special && (
+          <span
+            className={classNames('font-normal text-neutral-500 line-through', classNameVariants.special)} // [!code --]
+            className={classNames('font-normal text-neutral-500 line-through')} // [!code ++]
+            data-testid="regular-price"
+          >
+            {special}
+          </span>
+        )}
+      </div>
+    )
+  );
+}
+```
+
+Then, add the prop only to stores that need it (e.g., `store-a`):
+
+```tsx
+// /apps/stores/store-a/storefront-unified-nextjs/components/decorated-price.tsx
+
+import classNames from 'classnames';
+
+import { useFormatter } from '@/hooks/use-formatter';
+import type { SfProduct } from '@/types';
+
+export interface ClassNameVariants {
+  regular?: string;
+  special?: string;
+}
+
+// Added prop to store-a
+export interface DecoratedPriceProps {
+  className?: string;
+  classNameVariants: ClassNameVariants; // [!code ++]
```

Comment:
Now, I think we should suggest adding only optional fields in overrides. The reason is the following: when we add a required prop we have to override all the usages of the component as well. Otherwise, the app might break at runtime, throwing `Cannot read properties of undefined` or won't build with not so straightforward error. What's worse is when we add new usage of the component in the parent store, we introduce this bug unnoticeably. 

This is somewhat related to the Liskov Substitution Principle.

Reply:
Yeah, i've updated it.